test/built-ins/TypedArray/prototype/subarray/tointeger-begin.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2016 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-%typedarray%.prototype.subarray
+description: ToInteger(begin)
+info: >
+  22.2.3.27 %TypedArray%.prototype.subarray( begin , end )
+
+  ...
+  7. Let relativeBegin be ? ToInteger(begin).
+  ...
+includes: [testTypedArray.js, compareArray.js]
+---*/
+
+var obj = {
+  valueOf: function() {
+    return 2;
+  }
+};
+
+testWithTypedArrayConstructors(function(TA) {
+  var sample = new TA([40, 41, 42, 43]);
+
+  assert(compareArray(sample.subarray(false), [40, 41, 42, 43]), "false");
+  assert(compareArray(sample.subarray(true), [41, 42, 43]), "true");
+
+  assert(compareArray(sample.subarray(NaN), [40, 41, 42, 43]), "NaN");
+  assert(compareArray(sample.subarray(null), [40, 41, 42, 43]), "null");
+  assert(compareArray(sample.subarray(undefined), [40, 41, 42, 43]), "undefined");
+
+  assert(compareArray(sample.subarray(1.1), [41, 42, 43]), "1.1");
+  assert(compareArray(sample.subarray(0.1), [40, 41, 42, 43]), "0.1");
```

Comment:
These two assertions basically "prove" the same detail. A useful addition would be a value like `0.6` or `1.5`--these would prove that "floor" (not "round") is being used.

And on that note, a negative floating-point number would be useful here, too. Note how the algorithm converts values like `-1.1` to `-1`--a careless implementation might skip the absolute value part and convert to `-2` instead.